src/otx/algorithms/detection/adapters/mmdet/apis/train.py

```diff
@@ -119,11 +121,26 @@ def train_detector(model, dataset, cfg, distributed=False, validate=False, times
     elif cfg.device == "xpu":
         model = build_dp(model, cfg.device, device_ids=cfg.gpu_ids, enable_autocast=bool(fp16_cfg))
         model.to(f"xpu:{cfg.gpu_ids[0]}")
+    elif cfg.device == "hpu":
+        import habana_frameworks.torch.core as htcore
+        from habana_frameworks.torch.utils.library_loader import load_habana_module
+
+        load_habana_module()
+        os.environ["PT_HPU_LAZY_MODE"] = "1"
+        assert len(cfg.gpu_ids) == 1
+        model = build_dp(
+            model, cfg.device, device_ids=cfg.gpu_ids, dim=0, enable_autocast=bool(fp16_cfg), put_gt_on_device=False
+        )
+        model.to(model.src_device_obj)
```

Comment:
@eunwoosh Does it seem same with segmentation? If not, `model` can be `model.cuda()` in `build_dp`.

Reply:
Because `cfg.device` is `hpu`, `model.cuda()` isn't executed. So this code is necessary, I think.

Reply:
I double checked in segmentation, with `model.cuda()` on gpu migration, model's weights are converted to `hpu`.
`cfg.device` is not important in this situation because it is just for using `HPUDataParallel` in `dp_factory`.

Reply:
Below code is `bulid_dp` function in mmseg. As you can see, because device is `hpu`, `model.cuda()` isn't executed. Due to same reason, `build_dp` in mmdet doesn't execute `model.cuda()` in `build_dp` function. This is mmdet [code](https://github.com/open-mmlab/mmdetection/ddddddddblob/v2.28.2/mmdet/utils/util_distribution.py#L10).
<img width="327" alt="image" src="https://github.com/openvinotoolkit/training_extensions/assets/92974184/1c3cacd5-a889-4e8b-a59b-18ae201673e3">

Reply:
Yes, you're right. So, I suggest `model.cuda()` as `build_dp` args.
```python
model = build_dp(model.cuda(), cfg.device, ...) # cfg.device="hpu"
```

Reply:
I don't think so, because some of users want to use on CPU.

Reply:
I don't think so, because some of users want to use on CPU.

Reply:
Oh, it seems that we don't need both of them because `DataParallel` casts model's device ([link](https://github.com/pytorch/pytorch/blob/e9ebda29d87ce0916ab08c06ab26fd3766a870e5/torch/nn/parallel/data_parallel.py#L147-L148)).
Let's discuss it later.

Reply:
oh ok then we don't have to call explicitly. That's good news.